src/engine/Engine.h

```diff
@@ -130,7 +131,7 @@ class Engine : public IEngine, public SignalHandler::Signalable
       Preprocessor access.
     */
     bool preprocessingEnabled() const;
-    const Preprocessor *getPreprocessor();
+    Preprocessor *getPreprocessor();
```

Comment:
The issue is setSolutionValuesOfEliminatedNeurons() would remove variables from _eliminatedEquations once they are assigned. Should I rewrite the function so that it is const?

Reply:
Well, if there's a reason for it not to be const, then that's fine.
If there are other places in the code that also call getPreprocessor, and obviously these work when it is const, we can have two copies of the method - a const version and a volatile version.

Reply:
I think DnC is the only place where getPreprocessor is invoked.